dsl/src/main/java/tech/pegasys/peeps/node/GoQuorum.java

```diff
@@ -0,0 +1,169 @@
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.peeps.node;
+
+import tech.pegasys.peeps.util.DockerLogs;
+
+import java.util.List;
+
+import com.google.common.collect.Lists;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.AbstractWaitStrategy;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.MountableFile;
+
+public class GoQuorum extends Web3Provider {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  //  private static final String IMAGE_NAME = "hyperledger/besu:latest";
+  private static final String IMAGE_NAME = "quorumengineering/quorum";
+  private static final String CONTAINER_GENESIS_FILE = "/etc/genesis.json";
+  private static final String CONTAINER_NODE_PRIVATE_KEY_FILE = "/etc/keys/node.priv";
+  private static final String DATA_DIR = "/eth";
+
+  public GoQuorum(final Web3ProviderConfiguration config) {
+    super(config, new GenericContainer<>(IMAGE_NAME));
+    final List<String> commandLineOptions = standardCommandLineOptions();
+
+    addCorsOrigins(config, commandLineOptions);
+    addBootnodeAddress(config, commandLineOptions);
+    addContainerNetwork(config, container);
+    addContainerIpAddress(ipAddress(), container);
+    commandLineOptions.addAll(List.of("--datadir", "\"" + DATA_DIR + "\""));
+
+    container.withCopyFileToContainer(
+        MountableFile.forHostPath(config.getGenesisFile()), CONTAINER_GENESIS_FILE);
+    final List<String> entryPoint = Lists.newArrayList("/bin/sh", "-c");
+    final String initCmd =
+        "mkdir -p '"
+            + DATA_DIR
+            + "/geth' && geth --datadir \""
+            + DATA_DIR
+            + "\" init "
+            + CONTAINER_GENESIS_FILE
+            + " && echo '##### GoQuorum INITIAILSED #####' && ";
+
+    addNodePrivateKey(config, commandLineOptions, container);
+    //    if (config.isPrivacyEnabled()) {
+    //      addPrivacy(config, commandLineOptions, dockerContainer);
+    //    }
+
+    final String goCommandLine = initCmd + "geth " + String.join(" ", commandLineOptions);
+    LOG.info("GoQuorum command line: {}", goCommandLine);
+
+    entryPoint.add(goCommandLine);
+
+    container
+        .withCreateContainerCmdModifier(cmd -> cmd.withEntrypoint(entryPoint))
+        .waitingFor(liveliness());
+  }
+
+  @Override
+  public String getLogs() {
+    return DockerLogs.format("GoQuorum", container);
+  }
+
+  private AbstractWaitStrategy liveliness() {
+    return Wait.forLogMessage(".*endpoint=0.0.0.0:8545.*", 1);
+  }
+
+  private List<String> standardCommandLineOptions() {
+    return Lists.newArrayList(
+        "--verbosity",
+        "5",
+        "--rpccorsdomain",
+        "\"*\"",
+        // "--mine",
```

Comment:
Would be good for GoQuorum to be configured as a miner same as the Besu node

Reply:
Yes it would, but that requires Wallets and things, which I think can be moved to the next ticket.